honeybee_radiance_command/dcglare.py

```diff
@@ -0,0 +1,186 @@
+        self.dctotal = dctotal
+        self.sky_matrix = sky_matrix
+        self.vmtx = vmtx
+        self.dmtx = dmtx
+        self.tmtx = tmtx
+
+    @property
+    def options(self):
+        """dcglare options."""
+        return self._options
+
+    @options.setter
+    def options(self, value):
+        if not value:
+            value = DcglareOptions()
+
+        if not isinstance(value, DcglareOptions):
+            raise ValueError('Expected Dcglare options not {}'.format(value))
+
+        self._options = value
+    
+    @property
+    def dcdirect(self):
+        """Direct contribution matrix."""
+        return self._dcdirect
+    
+    @dcdirect.setter
+    def dcdirect(self, value):
+        if value is None:
+            self._dcdirect = None
+        else:
+            self._dcdirect = typing.normpath(value)
+    
+    @property
+    def dctotal(self):
+        """Total (direct and diffuse) contribution matrix."""
+        return self._dctotal
+    
+    @dctotal.setter
+    def dctotal(self, value):
+        if value is None:
+            self._dctotal = None
+        else:
+            self._dctotal = typing.normpath(value)
+    
+    @property
+    def sky_matrix(self):
+        """Sky contribution matrix."""
+        return self._sky_matrix
+    
+    @sky_matrix.setter
+    def sky_matrix(self, value):
+        if value is None:
+            self._sky_matrix = None
+        else:
+            self._sky_matrix = typing.normpath(value)
+
+    @property
+    def vmtx(self):
+        """View matrix."""
+        return self._vmtx
+    
+    @vmtx.setter
+    def vmtx(self, value):
+        if value is None:
+            self._vmtx = None
+        else:
+            self._vmtx = typing.normpath(value)
+
+    @property
+    def dmtx(self):
+        """Daylight matrix."""
+        return self._dmtx
+    
+    @dmtx.setter
+    def dmtx(self, value):
+        if value is None:
+            self._dmtx = None
+        else:
+            self._dmtx = typing.normpath(value)
+
+    @property
+    def tmtx(self):
+        """Transmission matrix (BSDF)."""
+        return self._tmtx
+    
+    @tmtx.setter
+    def tmtx(self, value):
+        if value is None:
+            self._tmtx = None
+        else:
+            self._tmtx = typing.normpath(value)
+
+    def to_radiance(self, stdin_input=False):
+        """Command in Radiance format.
+
+        Args:
+            stdin_input: A boolean that indicates if the input for this command
+                comes from stdin. This is for instance the case when you pipe the input
+                from another command. (Default: False).
+        """
+        self.validate(stdin_input)
+
+        command_parts = [self.command, self.options.to_radiance()]
+        command_parts += [self.dcdirect]
+        command_parts += [self.dctotal] if not self.tmtx else [self.vmtx, self.tmtx,
+                          self.dmtx]
+        command_parts += [self.sky_matrix] if self.sky_matrix else []
```

Comment:
Why don't you simply append these values to the list instead of creating a new list every time?

Reply:
I was wrong! `+=` adds the item to the existing list and this is totally fine.